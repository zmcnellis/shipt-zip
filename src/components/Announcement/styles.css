/* Announcements -------------------------------------------------------------*/
.announcement {
  display: block;
  text-align: center;
  background-color: #414042;
  width: 100%;
  height: 1.75rem;
  text-decoration: none;
  color: #fff;
}

.announcement:hover {
  background-color: #71b045;
  transition: all 0.3s ease;
}

.announcement-text {
  margin: 0;
  line-height: 1.75rem;
  text-transform: uppercase;
}
