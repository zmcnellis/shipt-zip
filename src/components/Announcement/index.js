import React, { Component } from 'react';
import './styles.css';

class Announcement extends Component {
  render() {
    const text = "Get $50 Off + $15 Credit*";
    const url = "#";

    return (
      <a className="announcement" href={url}>
        <h5 className="announcement-text">{text}</h5>
      </a>
    );
  }
}

export default Announcement;
