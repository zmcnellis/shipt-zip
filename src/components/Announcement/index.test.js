import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Announcement from '.';

describe('Announcement', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Announcement/>
    );
    expect(output).toMatchSnapshot();
  });

  it('should display the correct text', () => {
    const output = shallow(
      <Announcement/>
    );
    expect(output.find('h5').hasClass('announcement-text')).toEqual(true);
  });

  it('should render on hover', () => {
    const output = shallow(
      <Announcement/>
    );
    output.simulate("mouseover");
    expect(output).toMatchSnapshot();
  });
});
