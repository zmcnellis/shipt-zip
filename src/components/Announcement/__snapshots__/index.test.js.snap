// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Announcement should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Announcement />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h5
        className="announcement-text"
>
        Get $50 Off + $15 Credit*
</h5>,
      "className": "announcement",
      "href": "#",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Get $50 Off + $15 Credit*",
        "className": "announcement-text",
      },
      "ref": null,
      "rendered": "Get $50 Off + $15 Credit*",
      "type": "h5",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h5
          className="announcement-text"
>
          Get $50 Off + $15 Credit*
</h5>,
        "className": "announcement",
        "href": "#",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get $50 Off + $15 Credit*",
          "className": "announcement-text",
        },
        "ref": null,
        "rendered": "Get $50 Off + $15 Credit*",
        "type": "h5",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Announcement should render on hover 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Announcement />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h5
        className="announcement-text"
>
        Get $50 Off + $15 Credit*
</h5>,
      "className": "announcement",
      "href": "#",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Get $50 Off + $15 Credit*",
        "className": "announcement-text",
      },
      "ref": null,
      "rendered": "Get $50 Off + $15 Credit*",
      "type": "h5",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h5
          className="announcement-text"
>
          Get $50 Off + $15 Credit*
</h5>,
        "className": "announcement",
        "href": "#",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get $50 Off + $15 Credit*",
          "className": "announcement-text",
        },
        "ref": null,
        "rendered": "Get $50 Off + $15 Credit*",
        "type": "h5",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
